### PR TITLE
Added & populated docs folder and readme file

### PR DESCRIPTION
**Pull request by dou9g**
To explain the AMI build process in greater detail.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+The Concourse pipeline job 'Upload-ami-buider-configs’ un-packs the new release and uploads the (config) files to an S3 bucket.
+Inside the bucket there is a Kafka folder and a DKS folder respectively.
+
+# REPO LOCATION
+https://github.com/dwp/ami-builder-configs
+https://github.ucds.io/dip/aws-management-infrastructure/blob/master/ci/jobs/build_amis/upload-ami-builder-configs.yml
+https://github.ucds.io/dip/aws-management-infrastructure/tree/master/ci/jobs/build_amis
+
+The config files will be downloaded from S3 via a provisioning.sh script by the 'ami_builder.py' script.
+Packer requires a Jinja2.template file which get converted to packer.json.
```
Comment (mark-r-m) [inline README.md]: get_s_

File: README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+# REPO LOCATION
+https://github.ucds.io/dip/aws-management-infrastructure/blob/master/packer.tf
+
+To trigger these builds manually from your terminal
+
+fly -t concourse check-resource -r management/aim-builder-configs-release
+fly -t concourse check-resource -r aws-crypto/dks-host-ami
+
+Eg. to trigger a check on a pipeline in concourse. 'fly -t concourse check-resource -r pipeline/aim-builder-configs-release’
```
Comment (mark-r-m) [inline README.md]: concourse, use 
also needs back ticks for `fly......release`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+# REPO LOCATION
+https://github.com/dwp/ami-builder/blob/master/ami_builder.py
+
+In order to configure the behaviour of the Lambda function, e.g. what source AMI to use, a payload file is required.
+Please see link for Packer docs https://www.packer.io/docs/builders/amazon.html
+
+As part of the ami-builder-configs-release Concourse pipeline job the 'ami-builder-configs-release'
+calls the Ami-builder Lambda function which includes the manifest payload file.
+It is here where Packer EC2 Instances are allowed HTTPS internet traffic outbound via NAT Gateway.
```
Comment (mark-r-m) [inline README.md]: via a NAT

File: README.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+https://github.ucds.io/dip/aws-management-infrastructure/blob/master/ci/jobs/build_amis/upload-ami-builder-configs.yml
+https://github.ucds.io/dip/aws-management-infrastructure/tree/master/ci/jobs/build_amis
+
+The config files will be downloaded from S3 via a provisioning.sh script by the 'ami_builder.py' script.
+Packer requires a Jinja2.template file which get converted to packer.json.
+
+# REPO LOCATION
+https://github.com/dwp/ami-builder/blob/master/ami_builder.py
+
+In order to configure the behaviour of the Lambda function, e.g. what source AMI to use, a payload file is required.
```
Comment (mark-r-m) [inline README.md]: we could link to a sample?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+It is here where Packer EC2 Instances are allowed HTTPS internet traffic outbound via NAT Gateway.
+
+CloudWatch logs can be found in /aws/lambda/ami_builder.
+
+# REPO LOCATION
+https://github.ucds.io/dip/aws-management-infrastructure/blob/master/packer.tf
+
+To trigger these builds manually from your terminal
+
+fly -t concourse check-resource -r management/aim-builder-configs-release
```
Comment (mark-r-m) [inline README.md]: needs a triple ticks
```
code block
```

File: README.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
+# ami-builder-configs
+Configuration files for building various AMIs using ami-builder
+
+Each time ami-builder-config code has a new commit & is merged, a new GitHub release is created and zipped
+This gets picked up by Concourse pipeline job 'Ami-builder-release' which is monitoring every hour looking for a new release.
+The Concourse pipeline order can be viewed in the docs folder for this repo.
```
Comment (steveburton4) [inline README.md]: I'm not sure what this bit means, when you say _concourse pipeline order_ what does that mean?

File: README.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
+# ami-builder-configs
+Configuration files for building various AMIs using ami-builder
+
+Each time ami-builder-config code has a new commit & is merged, a new GitHub release is created and zipped
+This gets picked up by Concourse pipeline job 'Ami-builder-release' which is monitoring every hour looking for a new release.
```
Comment (steveburton4) [inline README.md]: We could link to the pipeline in Concourse here?

File: README.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
+# ami-builder-configs
+Configuration files for building various AMIs using ami-builder
+
+Each time ami-builder-config code has a new commit & is merged, a new GitHub release is created and zipped
```
Comment (steveburton4) [inline README.md]: Is this via CircleCI? If so, probably worth explicitly stating that and maybe linking to CircleCI job?

File: README.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+This gets picked up by Concourse pipeline job 'Ami-builder-release' which is monitoring every hour looking for a new release.
+The Concourse pipeline order can be viewed in the docs folder for this repo.
+The reason this is done this way is you can't have push notifications from GitHub into Crown and there are rate limits for polling GitHub
+The Concourse pipeline job 'Upload-ami-buider-configs’ un-packs the new release and uploads the (config) files to an S3 bucket.
+Inside the bucket there is a Kafka folder and a DKS folder respectively.
```
Comment (steveburton4) [inline README.md]: I'd probably re-word this for clarity to something like "Inside the bucket there is a folder for each specific AMI - currently there are Kafka and DKS AMIs". If that's right of course :)

**Comment by steveburton4** [comment]
@dou9g - I think we might need some more detail in this readme. I'd probably approach it as a step by step guide to how the process works at a high level - maybe structure the doc like a walkthrough for creating a new AMI.

I.e. something like:

**To create a new AMI**
Step 1: Creating the AMI config and packer config
Step 2: Adding to CircleCI

And so on...

Those steps might be wrong by the way as I don't really know the process but that structure would help me a lot.

@mark-r-m what do you think about the above?

**Comment by mark-r-m** [comment]
> @dou9g - I think we might need some more detail in this readme. I'd probably approach it as a step by step guide to how the process works at a high level - maybe structure the doc like a walkthrough for creating a new AMI.
> 
> I.e. something like:
> 
> **To create a new AMI**
> Step 1: Creating the AMI config and packer config
> Step 2: Adding to CircleCI
> 
> And so on...
> 
> Those steps might be wrong by the way as I don't really know the process but that structure would help me a lot.
> 
> @mark-r-m what do you think about the above?

Totally @steveburton4 @dou9g

File: README.md [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+echo "ghr -t GITHUB_TOKEN -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME} -c ${CIRCLE_SHA1} -delete ${VERSION} ./artifacts/"
+ghr -t ${GITHUB_TOKEN} -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME} -c ${CIRCLE_SHA1} -delete ${VERSION} ./artifacts/
+
+```
+
+Step 2: Create an AWS Lambda function.
+
+[Packer Lambda Code](https://github.ucds.io/dip/aws-management-infrastructure/blob/master/packer.tf)
+Creating the Lambda function
+
```
Comment (benjaminsherwood) [inline README.md]: I would probably note, just for the sake of having done so, that the following code is Terraform examples.

File: README.md [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+Step 4: Create an S3 Bucket for Configuration Items
+As you can see from the payload file above, the Lambda function expects you to
+have an S3 bucket that contains the packer generic configuration template key file and
+(optionally) an S3 bucket containing any provisioning scripts and associated
+deployable artifacts that those scripts reference (the same bucket can be used
+for both, of course)
+This allows the Lambda itself to be completely generic, and to be used to
+generate arbitrarily complex AMIs, providing, of course, that the build can
+complete within the Lambda's defined execution time.
```
Comment (benjaminsherwood) [inline README.md]: Probably worth noting that this execution time is configured to be Lambda's maximum, of 15 minutes. We've not had one exceed this yet but eventually we may.

File: README.md [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+
+#CONCOURSE AMI BUILD PIPELINE JOBS:
+
+[AMI Builder Release](https://concourse.service.dw/teams/dataworks/pipelines/management)
+[AMI Builder Config Release](https://concourse.service.dw/teams/dataworks/pipelines/management?groups=AMIs)
+
+Below are some fly command line examples to trigger these builds manually from your terminal
+
+```
+fly -t concourse check-resource -r management/aim-builder-configs-release
```
Comment (tea-short) [inline README.md]: **aim**-builder-configs-release -> **ami**-builder-configs-release

File: README.md [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+```
+```
+fly -t concourse check-resource -r aws-crypto/dks-host-ami
+```
+
+Eg to trigger a check on a pipeline in concourse, use
+```
+fly -t concourse check-resource -r pipeline/aim-builder-configs-release
```
Comment (tea-short) [inline README.md]: **aim**-builder-configs-release -> **ami**-builder-configs-release

File: README.md (hunk 7 above)
Comment (dou9g) [reply README.md]: done

File: README.md (hunk 8 above)
Comment (dou9g) [reply README.md]: changed as requested

File: README.md [hunk 14]
```diff
@@ -0,0 +1,187 @@
...
+            LOG_LEVEL: DEBUG
+            AMI_NAME: dks-host-ami
+            INSTANCE_TYPE: m4.large
+            PACKER_TEMPLATE_KEY: ami-builder/dks-host/generic_packer_template.json.j2
+            PROVISION_SCRIPT_KEYS: '["ami-builder/dks-host/dks-host-install.sh"]'
+            PROVISION_FILE_KEYS: '["ami-builder/dks-host/dks","ami-builder/dks-host/dks.sh","ami-builder/dks-host/server.properties","ami-builder/dks-host/dks.logrotate"]'
+      - .: (( inject meta.plan.build-ami ))
+```
+
+#CONCOURSE AMI BUILD PIPELINE JOBS:
```
Comment (steveburton4) [inline README.md]: Think you need a space after the # here

Comment (dou9g) [reply README.md]: as requested space added

**Pull request merged**